### PR TITLE
Add cfg file definition conventions

### PR DESCRIPTION
**Pull request by chunfuwen**
In order to provide one consistent, human-readable, maintainable test case names, it is necessary to
apply some conventions rules to Cartesian Configuration format.

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
@santwana 
@smitterl 
@kylazhang 
@dzhengfy 
Welcome inputs and thoughts.

File: cfg_file_definition_convention.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+       - current
+   variants:
+       - cold_plug:
+       - hot_plug:
+   variants:
+       -positive_test:
+           status_error = "no"
+       -negative_test:
+           status_error = "yes"
+
```
Comment (chloerh) [inline cfg_file_definition_convention.rst:57]: Currently we have fewer cases that need to be constructed from matrix, but more cases with different structures and sets of parameters. And they usually look like this in our test cfg files:

```
- positive_test:
    variants:
        - test_1:
        - test_2:
- negative_test:
    variants:
        - n_test_1:
        - n_test_2:
```
The advantage of this way is that we only need to place test cases to the correct part (positive or negative) without having to concern about the whole structure of the cfg file which influences all its test cases.  And no need to put a lot of cases' patterns under '-positive_test' or '-negative_test' like those matrix cfg files: virsh_attach_detach_disk_matrix.cfg  or virsh_attach_detach_interface_matrix.cfg  with one single line that has 2153 characters just because it's filled with only xxx, xxx, xxx ...

Comment (kylazhang) [reply cfg_file_definition_convention.rst:57]: Agree

**Comment by kylazhang** [comment]
@chunfuwen I think you can add more for some tips you ever mentioned for condition in cfg  and parmas for specific cases in cfg

File: cfg_file_definition_convention.rst (hunk 1 above)
Comment (smitterl) [reply cfg_file_definition_convention.rst:57]: +1

File: cfg_file_definition_convention.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
+Cartesian configuration files(Cfg) will define test case names.
```
Comment (smitterl) [inline cfg_file_definition_convention.rst:1]: "files(Cfg)" -> "files (.cfg)"

File: cfg_file_definition_convention.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+How cfg works? please refer to https://avocado-vt.readthedocs.io/en/latest/CartesianConfig.html
```
Comment (smitterl) [inline cfg_file_definition_convention.rst:2]: "How cfg works? please" -> "How do cfg files work? Please"

File: cfg_file_definition_convention.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+In order to provide one consistent, human-readable, maintainable test case names, it is necessary to
```
Comment (smitterl) [inline cfg_file_definition_convention.rst:3]: "one [..] names" -> "[..] names"

File: cfg_file_definition_convention.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+           variants:
+               - use_auth_uuid:
+                   use_auth_uuid = "yes"
+               - use_auth_usage:
+                   use_auth_usage = "yes"
+
+
+4. Variant relationship will define case name order. Variant precedence is evaluated in top-down. In other words, 
+   the last parsed variants has precedence over earlier definitions,therefore the lowest variant name will be placed at the beginning
+   of whole test case name. The higher level one variant represents, the more lower its position should be placed in. for example:
```
Comment (smitterl) [inline cfg_file_definition_convention.rst:40]: "more lower" -> "lower"

**Review by smitterl — CHANGES_REQUESTED** [review]
Thank you @chunfuwen Could you check some minor grammatical issues I think to have spotted?

File: cfg_file_definition_convention.rst (hunk 2 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:1]: updated

File: cfg_file_definition_convention.rst (hunk 3 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:2]: updated

File: cfg_file_definition_convention.rst (hunk 4 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:3]: updated

File: cfg_file_definition_convention.rst (hunk 5 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:40]: updated

File: cfg_file_definition_convention.rst (hunk 1 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:57]: updated

File: cfg_file_definition_convention.rst (hunk 3 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst]: updated

File: cfg_file_definition_convention.rst (hunk 1 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst:57]: updated

File: cfg_file_definition_convention.rst (hunk 2 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst]: updated

File: cfg_file_definition_convention.rst (hunk 4 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst]: updated

File: cfg_file_definition_convention.rst (hunk 5 above)
Comment (chunfuwen) [reply cfg_file_definition_convention.rst]: updated

**Review by chloerh — APPROVED** [review]

**Comment by chunfuwen** [comment]
@smitterl 
@kylazhang 
comments updated accordingly ,please take a look.

**Review by kylazhang — APPROVED** [review]
